src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNameParser.cs

```diff
@@ -170,5 +171,22 @@ internal static bool TryParseAnonymousTypeParameterName(string typeParameterName
             propertyName = null;
             return false;
         }
+
+        // <ContainingFile>FN__ClassName`OptionalArity
+        private static readonly Regex s_fileTypeOrdinalPattern = new Regex(@">F(\d)+__", RegexOptions.Compiled);
+
+        internal static bool TryParseFileTypeName(string unmangledTypeName, out int ordinal, out string originalTypeName)
+        {
+            if (s_fileTypeOrdinalPattern.Match(unmangledTypeName) is Match { Success: true, Groups: var groups, Length: var prefixEndsAt }
```

Comment:
>if ( [](http://example.com/codeflow?start=12&length=4)

Should we check `name.StartsWith("<")`, to match similar `TryParse*` methods above?